### PR TITLE
Update mario_rl_tutorial.py

### LINKED ISSUE #1620: RuntimeError: CUDA error: out of memory while trying mario_rl_tutorial.py
When I tried the mario_rl_tutorial.py using CUDA, I received an out of memory error:
```
RuntimeError: CUDA error: out of memory
```
Monitoring the GPU memory I saw a constant incremental of the use of its memory, specially when the Agent starts the training. As a workaround I changed the cache method of the Agent, as it seems that all the experiencies are sent to the GPU:
```python
        if self.use_cuda:
            state = torch.tensor(state).cuda()
            next_state = torch.tensor(next_state).cuda()
            action = torch.tensor([action]).cuda()
            reward = torch.tensor([reward]).cuda()
            done = torch.tensor([done]).cuda()
 (...)

        self.memory.append((state, next_state, action, reward, done,))
```
Instead of that, I modified this piece for using CPU and changed the recall method, to send at this moment the batch of experiencies to GPU:
```python

    def recall(self):
        """
        Retrieve a batch of experiences from memory
        """
        batch = random.sample(self.memory, self.batch_size)
        state, next_state, action, reward, done = map(torch.stack, zip(*batch))
        return state.cuda(), next_state.cuda(), action.squeeze().cuda(), reward.squeeze().cuda(), done.squeeze().cuda()
```

Not sure if this is the right approach, but it works. However, it might impact on the learning performance.

Could anyone help here?

cc @vmoens @nairbv

### PR DESCRIPTION
**Pull request by neuralninja27**
Fixes #1620 

## Description
Modified code to remove the CUDA out of memory error

## Checklist

- [x] The issue that is being fixed is referred in the description (see above "Fixes #1620 ")
- [x] Only one issue is addressed in this pull request
- [x] Labels from the issue that this PR is fixing are added to this pull request
- [x] No unnessessary issues are included into this pull request.

cc @vmoens @nairbv

### CODE REVIEW

File: intermediate_source/mario_rl_tutorial.py [hunk 1]
```diff
@@ -348,7 +348,7 @@ def act(self, state):
 class Mario(Mario):  # subclassing for continuity
     def __init__(self, state_dim, action_dim, save_dir):
         super().__init__(state_dim, action_dim, save_dir)
-        self.memory = deque(maxlen=100000)
+        self.memory = deque(maxlen=60000) # Please change this value according to your GPU memory (current value is suitable for a GPU with 16GB RAM)
```
Comment (vmoens) [inline intermediate_source/mario_rl_tutorial.py:351]: @svekars 
TorchRL replay buffer won't sit on CUDA and be more efficient than a deque.
Not sure if we want to integrate it here?

**Comment by neuralninja27** [comment]
@svekars sure I'll look into it

File: intermediate_source/mario_rl_tutorial.py [hunk 2]
```diff
@@ -1,3 +1,4 @@
+, storage
```
Comment (vmoens) [inline intermediate_source/mario_rl_tutorial.py:1]:
```suggestion
```

**Review by vmoens — APPROVED** [review]
LGTM thanks for this!

**Pull request merged**